src/main/java/com/qualityraven/imagetester/api/ImageTester.java

```diff
@@ -49,22 +49,19 @@ public ResultCode execute() throws IOException {
             }
         }
         Process process = Runtime.getRuntime().exec(args.toArray(new String[0]));
-        final BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
-        String line;
         ResultCode resultCode = null;
-        while ((line = reader.readLine()) != null) {
-            if (line.contains("[New]") || line.contains("[Passed]")) {
-                resultCode = ResultCode.SUCCESS;
-                break;
-            }
-            if (line.contains("[Mismatch]")) {
-                resultCode = ResultCode.FAIL;
-                break;
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()))) {
+            String line;
+            while (resultCode == null && (line = reader.readLine()) != null) {
```

Comment:
The way the while loop is set up, as soon as `resultCode` is set to something, this `while` loop will exit, then return the stored resultCode (if it's always null, then it later sets to `ResultCode.EXECUTION_ERROR`, and then returns that)

My recommendation would be to not even create a `ResultCode` object (new line 52). Instead, return the ResultCode as soon as you can determine whether it's passed or failed.

After the while loop (i.e. you never found `[New]`, `[Passed]`, or `[Mismatch]`, then you directly `return ResultCode.EXECUTION_ERROR` since that's the last possibility here.

This also will simplify the `while` loop's conditional because you only need to make sure `line` isn't null

Reply:
Sounds good.